extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/quantiles/DoublesSketchComplexMetricSerde.java

```diff
@@ -0,0 +1,98 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.quantiles;
+
+import com.yahoo.memory.Memory;
+import com.yahoo.sketches.quantiles.DoublesSketch;
+import com.yahoo.sketches.quantiles.UpdateDoublesSketch;
+import io.druid.data.input.InputRow;
+import io.druid.segment.column.ColumnBuilder;
+import io.druid.segment.data.GenericIndexed;
+import io.druid.segment.data.ObjectStrategy;
+import io.druid.segment.serde.ComplexColumnPartSupplier;
+import io.druid.segment.serde.ComplexMetricExtractor;
+import io.druid.segment.serde.ComplexMetricSerde;
+
+import java.nio.ByteBuffer;
+
+public class DoublesSketchComplexMetricSerde extends ComplexMetricSerde
+{
+
+  private static final DoublesSketchObjectStrategy strategy = new DoublesSketchObjectStrategy();
+
+  @Override
+  public String getTypeName()
+  {
+    return DoublesSketchModule.DOUBLES_SKETCH;
+  }
+
+  @Override
+  public ObjectStrategy<DoublesSketch> getObjectStrategy()
+  {
+    return strategy;
+  }
+
+  @Override
+  public ComplexMetricExtractor getExtractor()
+  {
+    return new ComplexMetricExtractor()
+    {
+      @Override
+      public Class<?> extractedClass()
+      {
+        return DoublesSketch.class;
+      }
+
+      @Override
+      public Object extractValue(final InputRow inputRow, final String metricName)
+      {
+        final Object object = inputRow.getRaw(metricName);
+        if (object instanceof String) {
+          String objectString = (String) object;
+          if (Character.isDigit((objectString).charAt(0))) {
+            try {
+              Double doubleValue = Double.parseDouble(objectString);
+              UpdateDoublesSketch sketch = DoublesSketch.builder().setK(2).build();
+              sketch.update(doubleValue);
+              return sketch;
+            }
+            catch (NumberFormatException e) {
+              // Log.info("Expected Double. Got string with value " +
+              // objectString);
+            }
+          }
+        }
+
+        if (object == null || object instanceof DoublesSketch || object instanceof Memory) {
+          return object;
+        }
+        return DoublesSketchOperations.deserialize(object);
+      }
+    };
+  }
+
+  @Override
+  public void deserializeColumn(final ByteBuffer buffer, final ColumnBuilder builder)
+  {
+    final GenericIndexed<DoublesSketch> column = GenericIndexed.read(buffer, strategy);
```

Comment:
use `GenericIndexed.read(buffer, strategy, builder.getFileMapper())` similar to thetaSketch to enable largeColumns. Also Override `getSerializer` method as 
```
  @Override
  public GenericColumnSerializer getSerializer(IOPeon peon, String column)
  {
    return LargeColumnSupportedComplexColumnSerializer.create(peon, column, this.getObjectStrategy());
  }
```